tests/tests.cc

```
@@ -1539,7 +1539,40 @@ class Tests {
+    if (url.host != c.host || url.port != c.port) {
+      throw std::runtime_error(
+          "TestUrlParse(): Url::Parse(\"" + c.input + "\"): expected host='" +
+          c.host + "' port=" + std::to_string(c.port) + "; got host='" +
+          url.host + "' port=" + std::to_string(url.port));
+    }
+  }
+}
+
 int main(int /*argc*/, char* /*argv*/[]) {
+  // Unit check first so a parsing regression fails fast without a server.
+  TestUrlParse();
```

Cppcheck (2.21.0)

[error] Unhandled exception thrown in function that is an entry point.

(throwInEntryPoint)